### ISSUE TITLE
Электронные книги: генерировать epub-файл для всей книги R:AtZ

### ISSUE THREAD

**Issue by niya3** [lesswrong-ru/lesswrong-ru#91]
На странице https://lesswrong.ru/w сейчас есть только _fb2/mobi_ файлы для книги R:AtZ. epub-файла со всей книгой нет, лишь с отдельными (не всеми, #90) цепочками. Те новые пользователи, которые желают читать именно в этом формате, находят на просторах интернета сторонние, устаревшие версии от января 2019 года и пропускают все вышедшие с того момента исправления и обновления перевода книги. Стоит ценерировать epub и для всей книги целиком.

![image](https://user-images.githubusercontent.com/1140412/89822859-23f43280-db59-11ea-86c6-0da83601efef.png)

**Comment by berekuk** [comment]
Спасибо за этот и другие тикеты, я пока что напишу общий коммент на все.

К сожалению,с генерацией электронных книг на сайте всё странно, там система из костылей в виде bash-скрипта, python-обзязок и cli-интерфейса к calibre, которую примерно никто из текущей команды не понимает, и сейчас нет никого, кто был бы готов потратить ~1-2 дня времени, чтобы нормально разобраться.

Код этой системы лежит в отдельном репозитории, кстати: https://github.com/lesswrong-ru/generate-books

**Comment by niya3** [comment]
Да, я пробовал его ковырнуть даже как-то. В принципе, можно в нём можно разобраться и отрефакторить, был бы доступ до копий директорий с сервера, к которым этот скрипт обращается.

А вообще, есть ли статистика по скачиваниям? Насколько популярны электронные книги? Может никто их не качает, тогда и исправлять ничего не надо :)

**Comment by berekuk** [comment]
Судя по тому, что про них периодически репортят баги, всё же кто-то ими пользуется.

За ~9 дней, со 2 по 11 августа:

```
root@lesswrong-ru-new:/var/log/nginx# fgrep '.epub' access.log.1 | wc -l
123
root@lesswrong-ru-new:/var/log/nginx# fgrep '.fb2' access.log.1 | wc -l
52
root@lesswrong-ru-new:/var/log/nginx# fgrep '.mobi' access.log.1 | wc -l
3
```

Это всего запросов.

Если считать уникальные IP, то:

```
root@lesswrong-ru-new:/var/log/nginx# fgrep '.epub' access.log.1 | awk '{print $1}' | sort | uniq -c | wc -l
5
root@lesswrong-ru-new:/var/log/nginx# fgrep '.fb2' access.log.1 | awk '{print $1}' | sort | uniq -c | wc -l
12
root@lesswrong-ru-new:/var/log/nginx# fgrep '.mobi' access.log.1 | awk '{print $1}' | sort | uniq -c | wc -l
3
```

**Comment by berekuk** [comment]
> был бы доступ до копий директорий с сервера, к которым этот скрипт обращается

А он куда-то обращается к файлам? Я думал, он ходит на веб-урлы для экспорта.

**Comment by berekuk** [comment]
Если что, я только что запушил в https://github.com/lesswrong-ru/generate-books текущую версию кода (так вышло, что там часто правили код прямо на сервере).

На кроне стоят makebook-new.sh и makebook-sequences.sh, насколько я вижу.

**Comment by niya3** [comment]
как минимум, есть чтение файлов

* /srv/lesswrong.ru/files/lw.png
* /srv/lesswrong.ru/files/lwlogo.jpg
* /srv/lesswrong.ru/files/lwlogo.gif

**Comment by berekuk** [comment]
Ага, эти файлы можно получить просто по адресам https://lesswrong.ru/files/lw.png и т.д., весь lesswrong.ru хостится из директории /srv/lesswrong.ru.
